elf_reader.go

```diff
@@ -140,6 +141,12 @@ func LoadCollectionSpecFromReader(rd io.ReaderAt) (*CollectionSpec, error) {
 		return nil, fmt.Errorf("load data sections: %w", err)
 	}
 
+	ec.maps = &maps
```

Comment:
Please factor `maps` out from being a parameter on the methods above (in a separate commit) and initialize it when allocating the `elfCode` above. This is a no go, sorry.